examples/openvino_model/resnet50_client_acc.py

```diff
@@ -0,0 +1,193 @@
+# Copyright 2021 ZTE corporation. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+This is a sample for Adlik_serving prediction
+"""
+
+import argparse
+import os
+from adlik_serving import PredictContext
+import numpy as np
+import cv2  # #for resnet50
+# #for resnet101
+# from tensorflow.keras.preprocessing import image
+FLAGS = None
+
+
+def _parse_model(config, model_name, batch_size):
+    if config.max_batch_size == 0:
+        if batch_size != 1:
+            raise Exception("batching not supported for model '" + model_name + "'")
+    else:  # max_batch_size > 0
+        if batch_size > config.max_batch_size:
+            raise Exception(
+                "expecting batch size <= {} for model '{}'".format(config.max_batch_size, model_name))
+
+    input_ = config.input[0]
+    output = config.output[0]
+
+    return input_.name, output.name
+
+
+def _gen_input_data():
+    if os.path.isdir(FLAGS.image_filename):
+        file_names = [os.path.join(FLAGS.image_filename, f)
+                      for f in os.listdir(FLAGS.image_filename)
+                      if os.path.isfile(os.path.join(FLAGS.image_filename, f))]
+    else:
+        file_names = [FLAGS.image_filename]
+
+    file_names.sort()
+
+    image_data = []
+    for filename in file_names:
+        array = _preprocess(filename)
+        image_data.append(array)
+
+    true_labels = None
+    if FLAGS.true_label is not None:
+        with open(FLAGS.true_label, 'r') as fid:
+            true_label_lines = fid.read().replace('\r', '').split('\n')
+            true_labels = {}
+            for line in true_label_lines:
+                if len(line) and ' ' in line:
+                    key_val = line.split(' ')
+                    true_labels[key_val[0]] = int(key_val[1])
+
+    return file_names, image_data, true_labels
+
+# #for resnet 50
+
+
+def _preprocess(image_path):
+    def center_crop(im, crop_size):
+        height, width, channel = im.shape
+        w_start = (width - crop_size) // 2
+        h_start = (height - crop_size) // 2
+        w_end = w_start + crop_size
+        h_end = h_start + crop_size
+        return im[h_start:h_end, w_start:w_end, :]
+
+    def normalize(im, mean, std):
+        mean = np.array(mean)[np.newaxis, np.newaxis, :]
+        std = np.array(std)[np.newaxis, np.newaxis, :]
+        im = im / 255.0
+        im -= mean
+        im /= std
+        return im
+
+    im = cv2.imread(image_path).astype('float32')
+    im = cv2.cvtColor(im, cv2.COLOR_BGR2RGB)
+    im = cv2.resize(im, (256, 256), interpolation=cv2.INTER_LINEAR)
+    im = center_crop(im, 224)
+    im = normalize(im, [0.485, 0.456, 0.406], [0.229, 0.224, 0.225])
+    # im = im.transpose((2, 0, 1))
+    return im
+# #for resnet 101
```

Comment:
why resnet101 is different from resnet50 in preprocess? I think the only difference is model structure but not data preprocess method